tick/survival/simu_coxreg.py

```diff
@@ -202,3 +215,274 @@ def _as_dict(self):
+            The simulated (or given) features matrix
+
+        times : `numpy.ndarray`, shape=(n_samples,)
+            Simulated times
+
+        censoring : `numpy.ndarray`, shape=(n_samples,)
+            Simulated censoring indicator, where ``censoring[i] == 1``
+            indicates that the time of the i-th individual is a failure
+            time, and where ``censoring[i] == 0`` means that the time of
+            the i-th individual is a censoring time
+        """
+        return SimuWithFeatures.simulate(self)
+
+    @property
+    def times_distribution(self):
+        return self._times_distribution
+
+    @times_distribution.setter
+    def times_distribution(self, val):
+        if val != "weibull":
+            raise ValueError("``times_distribution`` was not "
+                             "understood, try using 'weibull' instead")
+        self._set("_times_distribution", val)
+
+    @property
+    def shape(self):
+        return self._shape
+
+    @shape.setter
+    def shape(self, val):
+        if val <= 0:
+            raise ValueError("``shape`` must be strictly positive")
+        self._set("_shape", val)
+
+    @property
+    def scale(self):
+        return self._scale
+
+    @scale.setter
+    def scale(self, val):
+        if val <= 0:
+            raise ValueError("``scale`` must be strictly positive")
+        self._set("_scale", val)
+
+    def _simulate(self):
+        # The features matrix already exists, and is created by the
+        # super class
+        features = self.features
+        n_samples, n_features = features.shape
+        # Simulation of cut-points
+        n_cut_points = self.n_cut_points
+        n_cut_points_factor = self.n_cut_points_factor
+        sparsity = self.sparsity
+        if not 0 <= sparsity <= 1:
+            raise ValueError("``sparsity`` must be in (0, 1), "
+                             "got %s" % sparsity)
+        s = round(n_features * sparsity)
+        # sparsity index set
+        S = np.random.choice(n_features, s, replace=False)
+
+        if n_cut_points is None:
+            n_cut_points = np.random.geometric(n_cut_points_factor, n_features)
+        else:
+            n_cut_points = np.repeat(n_cut_points, n_features)
+
+        cut_points = {}
+        coeffs_binarized = np.array([])
+        for j in range(n_features):
+            feature_j = features[:, j]
+            quantile_cuts = np.linspace(10, 90, 10)
+            candidates = np.percentile(feature_j, quantile_cuts,
+                                       interpolation="nearest")
+            cut_points_j = np.random.choice(candidates, n_cut_points[j],
+                                            replace=False)
+            cut_points_j = np.sort(cut_points_j)
+            cut_points_j = np.insert(cut_points_j, 0, -np.inf)
+            cut_points_j = np.append(cut_points_j, np.inf)
+            cut_points[str(j)] = cut_points_j
+            # generate beta star
+            if j in S:
```

Comment:
There's no choice in the way we build coefficients in this class ? It's hard coded here ?

Reply:
Yes, see comment above